yahttp build warnings

```
/bin/sh ../../../../libtool  --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I. -I../../../../../pdns/ext/yahttp/yahttp -I../../../..   -pthread  -I../../../.. -I../../../../.. -DRUBEN_CXXFLAGS -D_GNU_SOURCE -Wall -O2 -c -o reqresp.lo ../../../../../pdns/ext/yahttp/yahttp/reqresp.cpp
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I../../../../../pdns/ext/yahttp/yahttp -I../../../.. -pthread -I../../../.. -I../../../../.. -DRUBEN_CXXFLAGS -D_GNU_SOURCE -Wall -O2 -c ../../../../../pdns/ext/yahttp/yahttp/reqresp.cpp  -fPIC -DPIC -o .libs/reqresp.o
../../../../../pdns/ext/yahttp/yahttp/reqresp.cpp: In instantiation of 'int YaHTTP::AsyncLoader<T>::feed(const string&) [with T = YaHTTP::Response; std::string = std::basic_string<char>]':
../../../../../pdns/ext/yahttp/yahttp/reqresp.cpp:198:51:   required from here
../../../../../pdns/ext/yahttp/yahttp/reqresp.cpp:89:40: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
       if (target->kind == YAHTTP_TYPE_REQUEST && minbody > target->max_request_size) throw ParseError("Max request body size exceeded");
                                        ^
../../../../../pdns/ext/yahttp/yahttp/reqresp.cpp:90:45: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
       else if (target->kind == YAHTTP_TYPE_RESPONSE && minbody > target->max_response_size) throw ParseError("Max response body size exceeded");
                                             ^
../../../../../pdns/ext/yahttp/yahttp/reqresp.cpp: In instantiation of 'int YaHTTP::AsyncLoader<T>::feed(const string&) [with T = YaHTTP::Request; std::string = std::basic_string<char>]':
../../../../../pdns/ext/yahttp/yahttp/reqresp.cpp:221:51:   required from here
../../../../../pdns/ext/yahttp/yahttp/reqresp.cpp:89:40: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
       if (target->kind == YAHTTP_TYPE_REQUEST && minbody > target->max_request_size) throw ParseError("Max request body size exceeded");
                                        ^
../../../../../pdns/ext/yahttp/yahttp/reqresp.cpp:90:45: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
       else if (target->kind == YAHTTP_TYPE_RESPONSE && minbody > target->max_response_size) throw ParseError("Max response body size exceeded");
                                             ^
```

--- a/pdns/ext/yahttp/yahttp/reqresp.cpp
+++ b/pdns/ext/yahttp/yahttp/reqresp.cpp
@@ -86,8 +86,8 @@ namespace YaHTTP {
         maxbody = minbody;
       }
       if (minbody < 1) return true; // guess there isn't anything left.
-      if (target->kind == YAHTTP_TYPE_REQUEST && minbody > target->max_request_size) throw ParseError("Max request body size exceeded");
-      else if (target->kind == YAHTTP_TYPE_RESPONSE && minbody > target->max_response_size) throw ParseError("Max response body size exceeded");
+      if (target->kind == YAHTTP_TYPE_REQUEST && static_cast<ssize_t>(minbody) > target->max_request_size) throw ParseError("Max request body size exceeded");
+      else if (target->kind == YAHTTP_TYPE_RESPONSE && static_cast<ssize_t>(minbody) > target->max_response_size) throw ParseError("Max response body size exceeded");
     }
 
     if (maxbody == 0) hasBody = false;

--- a/pdns/ext/yahttp/yahttp/router.hpp
+++ b/pdns/ext/yahttp/yahttp/router.hpp
@@ -34,9 +34,9 @@ namespace YaHTTP {
 
 This class implements a router for masked urls. The URL mask syntax is as of follows
 
-/<masked>/url<number>/<hi>.<format>
+/&lt;masked&gt;/url&lt;number&gt;/&lt;hi&gt;.&lt;format&gt;
 
-You can use <*param> to denote that everything will be matched and consumed into the parameter, including slash (/). Use <*> to denote that URL 
+You can use &lt;*param&gt; to denote that everything will be matched and consumed into the parameter, including slash (/). Use &lt;*&gt; to denote that URL 
 is consumed but not stored. Note that only path is matched, scheme, host and url parameters are ignored. 
    */
   class Router {

--- a/pdns/ext/yahttp/yahttp/utility.hpp
+++ b/pdns/ext/yahttp/yahttp/utility.hpp
@@ -224,7 +224,7 @@ namespace YaHTTP {
         if (!std::isalnum(*iter) && (!asUrl || skip.find(*iter) == std::string::npos)) {
           // replace with different thing
           pos = std::distance(result.begin(), iter);
-          snprintf(repl,3,"%02x", static_cast<unsigned char>(*iter));
+          ::snprintf(repl,3,"%02x", static_cast<unsigned char>(*iter));
           result = result.replace(pos, 1, "%", 1).insert(pos+1, repl, 2);
           iter = result.begin() + pos + 2;
         }

